### PR TITLE
Add tests with a jsoc cutout file and update MGN to convert int to float

### PR DESCRIPTION
**Pull request by nabobalis**
Fixes https://github.com/sunpy/sunkit-image/issues/211

Issue was reported with JSOC cutouts and the datatype breaking these routines.

### CODE REVIEW

File: sunkit_image/enhance.py [hunk 1]
```diff
@@ -101,51 +101,42 @@ def mgn(
...
     image = np.zeros_like(data)
     conv = np.zeros_like(data)
     sigmaw = np.zeros_like(data)
-
     for s, weight in zip(sigma, weights, strict=True):
         # 2 & 3 Create kernel and convolve with image
         # Refer to equation (1) in the paper
-        ndimage.gaussian_filter(data, sigma=s, truncate=truncate, mode="nearest", output=conv)
-
+        conv = ndimage.gaussian_filter(data, sigma=s, truncate=truncate, mode="nearest")
```
Comment (wtbarnes) [inline sunkit_image/enhance.py:116]: Why the change here? Presumably the `output` kwarg is used to avoid making a copy of the array.

File: sunkit_image/enhance.py [hunk 2]
```diff
@@ -101,51 +101,42 @@ def mgn(
...
         )
-
     if weights is None:
         weights = np.ones(len(sigma))
-
     # 1. Replace spurious negative pixels with zero
     if clip is True:
         data[data <= 0] = 1e-15  # Makes sure that all values are above zero
-
+    data = data.astype(np.float32) if data.dtype != np.float32 else data
```
Comment (wtbarnes) [inline sunkit_image/enhance.py:109]: Why float32? Why not double precision for example?

File: sunkit_image/enhance.py [hunk 3]
```diff
@@ -101,51 +101,42 @@ def mgn(
...
         # square the difference, and convolve with kernel. Square-root the
         # resulting image to give `local standard deviation` image sigmaw
         # Refer to equation (2) in the paper
         conv = data - conv
-        ndimage.gaussian_filter(conv**2, sigma=s, truncate=truncate, mode="nearest", output=sigmaw)
-        np.sqrt(sigmaw, out=sigmaw)
-
+        sigmaw = ndimage.gaussian_filter(conv**2, sigma=s, truncate=truncate, mode="nearest")
+        with np.errstate(invalid="ignore"):
+            sigmaw = np.sqrt(sigmaw)
```
Comment (wtbarnes) [inline sunkit_image/enhance.py:124]: Same comment as above. Why not make use of `output`?

File: sunkit_image/enhance.py (hunk 2 above)
Comment (nabobalis) [reply sunkit_image/enhance.py:109]: I just picked one at random. I should just promote it to that instead. That makes more sense.

File: sunkit_image/enhance.py (hunk 1 above)
Comment (nabobalis) [reply sunkit_image/enhance.py:116]: I just don't like it but yeah. I will undo this.

File: sunkit_image/enhance.py (hunk 2 above)
Comment (nabobalis) [reply sunkit_image/enhance.py:109]: Trying that I still get:

```
numpy._core._exceptions._UFuncOutputCastingError: Cannot cast ufunc 'sqrt' output from dtype('float64') to dtype('int64') with casting rule 'same_kind'
```

I can probably make it unsafe?

Comment (nabobalis) [reply sunkit_image/enhance.py]: I came up with worse ideas.

File: sunkit_image/asda.py [hunk 4]
```diff
@@ -1,391 +1,391 @@
-"""
```
Comment (nabobalis) [inline sunkit_image/asda.py:1]: This had the wrong last line character.

File: sunkit_image/granule.py [hunk 5]
```diff
@@ -1,282 +1,282 @@
-"""
```
Comment (nabobalis) [inline sunkit_image/granule.py:1]: This had the wrong last line character.

File: sunkit_image/tests/test_granule.py [hunk 6]
```diff
@@ -1,124 +1,124 @@
-import numpy as np
```
Comment (nabobalis) [inline sunkit_image/tests/test_granule.py:1]: This had the wrong last line character.

**Review by alasdairwilson — APPROVED** [review]

**Comment by alasdairwilson** [comment]
I don't count but +1

**Comment by wtbarnes** [comment]
What was the actual issue here? Do the cutouts from the JSOC have a different datatype than the full-disk L1 files?

**Comment by nabobalis** [comment]
> What was the actual issue here? Do the cutouts from the JSOC have a different datatype than the full-disk L1 files?

I didn't debug the reason.

File: sunkit_image/tests/test_granule.py
Comment (nabobalis) [inline sunkit_image/tests/test_granule.py:1]: This is jut a end of file character change.

File: sunkit_image/asda.py
Comment (nabobalis) [inline sunkit_image/asda.py:1]: This is jut a end of file character change.

File: sunkit_image/enhance.py [hunk 7]
```diff
@@ -101,51 +102,40 @@ def mgn(
...
         # Refer to equation (1) in the paper
         ndimage.gaussian_filter(data, sigma=s, truncate=truncate, mode="nearest", output=conv)
-
         # 4. Calculate difference between image and the local mean image,
         # square the difference, and convolve with kernel. Square-root the
         # resulting image to give `local standard deviation` image sigmaw
         # Refer to equation (2) in the paper
         conv = data - conv
         ndimage.gaussian_filter(conv**2, sigma=s, truncate=truncate, mode="nearest", output=sigmaw)
         np.sqrt(sigmaw, out=sigmaw)
```
Comment (nabobalis) [inline sunkit_image/enhance.py:123]:
```python
>           np.sqrt(sigmaw, out=sigmaw)
E           numpy._core._exceptions._UFuncOutputCastingError: Cannot cast ufunc 'sqrt' output from dtype('float32') to dtype('int16') with casting rule 'same_kind'
```
So this is the original issue that we need to fix.

**Review by wtbarnes — APPROVED** [review]
